cpp/src/arrow/util/logging.h

```diff
@@ -80,42 +80,36 @@ enum class ArrowLogLevel : int {
 #ifdef NDEBUG
 #define ARROW_DFATAL ::arrow::util::ArrowLogLevel::ARROW_WARNING
 
-#define DCHECK(condition)       \
-  ARROW_IGNORE_EXPR(condition); \
-  while (false) ::arrow::util::ArrowLogBase()
-#define DCHECK_OK(status)    \
-  ARROW_IGNORE_EXPR(status); \
-  while (false) ::arrow::util::ArrowLogBase()
+#define DCHECK(condition) \
+  while (false) ARROW_CHECK(condition)
+#define DCHECK_OK(status) \
+  while (false) ARROW_CHECK_OK(status)
 #define DCHECK_EQ(val1, val2) \
-  ARROW_IGNORE_EXPR(val1);    \
-  while (false) ::arrow::util::ArrowLogBase()
+  while (false) ARROW_CHECK((val1) == (val2))
 #define DCHECK_NE(val1, val2) \
-  ARROW_IGNORE_EXPR(val1);    \
-  while (false) ::arrow::util::ArrowLogBase()
+  while (false) ARROW_CHECK((val1) != (val2))
 #define DCHECK_LE(val1, val2) \
-  ARROW_IGNORE_EXPR(val1);    \
-  while (false) ::arrow::util::ArrowLogBase()
+  while (false) ARROW_CHECK((val1) <= (val2))
 #define DCHECK_LT(val1, val2) \
-  ARROW_IGNORE_EXPR(val1);    \
-  while (false) ::arrow::util::ArrowLogBase()
+  while (false) ARROW_CHECK((val1) < (val2))
 #define DCHECK_GE(val1, val2) \
-  ARROW_IGNORE_EXPR(val1);    \
-  while (false) ::arrow::util::ArrowLogBase()
+  while (false) ARROW_CHECK((val1) >= (val2))
 #define DCHECK_GT(val1, val2) \
-  ARROW_IGNORE_EXPR(val1);    \
-  while (false) ::arrow::util::ArrowLogBase()
+  while (false) ARROW_CHECK((val1) > (val2))
+#define DABORT_NOT_OK(expr) ARROW_IGNORE_EXPR(expr)
```

Comment:
It seems this does not guarantee that the statements are executed in release builds

Reply:
`ARROW_IGNORE_EXPR(expr)` evaluates the expression but cast it to void, so the compiler won't warn against a Status return which isn't examined